data_sources/data_source.py

```diff
@@ -0,0 +1,35 @@
+from abc import ABC, abstractmethod
+from typing import List, Dict
+
+import boto3
+from botocore.exceptions import NoCredentialsError
+import uuid
+import pandas as pd
+from io import BytesIO
+
+
+class DataSource(ABC):
+    def __init__(self, s3_bucket):
```

Comment:
Why s3 specifically? Shouldn't it be generic?

Reply:
we want any intermediate storage to be in s3